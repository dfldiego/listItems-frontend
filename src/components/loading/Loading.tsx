type Props = {
  loading: boolean;
};

const Loading = ({ loading }: Props) => {
  return (
    <>
      {loading && (
        <div
          className="spinner mt5 loadingGrid"
          data-testid="loading-test-id"
        ></div>
      )}
    </>
  );
};

export default Loading;
