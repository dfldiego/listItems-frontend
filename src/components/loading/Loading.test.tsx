import { render, screen } from "@testing-library/react";
import Loading from "./Loading";

const PROPS = {
  loading: true,
};

const PROPS_FALSE = {
  loading: false,
};

describe("<Loading>", () => {
  test("should render correctly", () => {
    const { container } = render(<Loading {...PROPS} />);
    expect(container).toMatchSnapshot();
  });

  test("should render the loading", () => {
    render(<Loading {...PROPS} />);
    const loading = screen.getByTestId("loading-test-id");
    expect(loading).toHaveClass("spinner");
  });

  test("should not render the error message", () => {
    render(<Loading {...PROPS_FALSE} />);
    expect(screen.queryByTestId("loading-test-id")).toBeFalsy();
  });
});
