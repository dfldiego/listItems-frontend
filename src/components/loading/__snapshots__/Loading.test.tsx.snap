// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Loading> should render correctly 1`] = `
<div>
  <div
    class="spinner mt5 loadingGrid"
    data-testid="loading-test-id"
  />
</div>
`;
